src/main/java/org/terracotta/build/plugins/DeployPlugin.java

```diff
@@ -178,17 +181,26 @@ public void execute(Task task) {
         java.withJavadocJar();
         java.withSourcesJar();
       });
+    });
 
-      project.afterEvaluate(p -> {
-        p.getExtensions().configure(PublishingExtension.class, publishing -> {
-          if (publishing.getPublications().isEmpty()) {
-            publishing.publications(publications -> publications.register("mavenJava", MavenPublication.class, mavenJava -> mavenJava.from(p.getComponents().getByName("java"))));
+    project.afterEvaluate(p -> {
+      p.getExtensions().configure(PublishingExtension.class, publishing -> {
+        if (publishing.getPublications().isEmpty()) {
```

Comment:
If publishing and no components have been marked, and there is a singular component available... publish it. This handles the common situations of Package or JavaLibrary plugin use, while allowing customization too.